monitoring/tests/test_finished_build_stats_publisher.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+import unittest
+import timeout_decorator
+from unittest.mock import MagicMock
+from finished_build_stats_publisher import FinishedBuildStatsPublisher, MaxRetriesExceededError
+from tests.common import get_test_build_stats_with_status
+
+
+class TestFinishedBuildStatsPublisher(unittest.TestCase):
+    def setUp(self):
+        self.collector = MagicMock()
+        self.publisher = MagicMock()
+        self.publisher.publish = MagicMock()
+        self.stats_publisher = FinishedBuildStatsPublisher(self.collector, self.publisher)
+        self.build_stats_success = get_test_build_stats_with_status('SUCCESS')
+        self.build_stats_in_progress = get_test_build_stats_with_status('IN_PROGRESS')
+
+    def test_finished(self):
```

Comment:
test_finished name is weak and will not survive the winter. Maybe something like test_collecting_already_finished_build would be better?